include/zephyr/arch/riscv/arch.h

```diff
@@ -29,6 +29,9 @@
 #include <zephyr/devicetree.h>
 #include <zephyr/arch/riscv/csr.h>
 #include <zephyr/arch/riscv/exception.h>
+#if defined(CONFIG_RISCV_ISA_EXT_A)
+#include <zephyr/arch/riscv/atomic.h>
```

Comment:
This change should probably be moved somewhere here: https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/sys/atomic.h#L30